docs/technical-design/file-upload.md

```diff
@@ -0,0 +1,93 @@
+# How FileUpload works on MC-Review
+
+## Introduction
+
+MC-Review allows users to upload and save files through multiple workflows. FileUpload is the main reusable component for file uploads in the application. It supports multiple files at a time via user select or drag and drop.
+
+Currently, the component is embedded inside several larger forms in our application. The document files supported in MC-Review are text based documents (we support  PDF, CSV, DOC, DOCX, XLS, XLSX, XLSM).
+
+## FileUpload 101
+
+### FileUpload is two pieces of user interface that live close together
+
+- **A file input.** The input can be triggered by click or by drag and drop with a valid file.
+The look and feel of the file input [draws directly from USWDS](https://designsystem.digital.gov/components/file-input/) (via the [shared component exported by react-uswds](https://github.com/trussworks/react-uswds/tree/main/src/components/forms/FileInput)).
+- **A file list.** This is the UI that shows up below the input. It is custom built for MC-Review. There are currently two variations of the file list that are supported - an HTML table and an HTML unordered list.
+
+### FileUpload has three main functions
+
+- it coordinates between the input and the file list UI
+- it keeps the file list stored in React state up to data with async AWS S3 requests
+- it communicates the overall state of files and any errors to the parent form component (which will display additional validation messages to the user when the form is in validation state)
+
+## Expected behavior
+
+A text summary of the expected behavior and the error states are below.
+
+### Happy path
+
+- **The  user selects or drops a file.** As soon as the user takes action, the application starts uploading the file to  AWS S3 in the background. The loading indicator is displayed.
```

Comment:
@haworku To the point you brought up regarding accessibility, this feels like a good spot to at least link to the Confluence page. I would assume anyone who has access to the Github has access to Confluence, correct?